File: examples/python/alldl.py
import requests

API_BASE = 'https://cog.api.br/api/v1'

def download_all_media(url: str) -> dict:
    """
    Download all available media from any URL
    
    Args:
        url: Any URL supported by yt-dlp
        
    Returns:
        dict: All media data or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/alldl', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            result = data['data']
            metadata = result['metadata']
            media = result['media']
            
            print('=== METADATA ===')
            print(f"Title: {metadata['title']}")
            print(f"Platform: {metadata['platform']}")
            print(f"Duration: {metadata['duration']} seconds")
            print(f"Views: {metadata.get('views', 'N/A')}")
            print(f"Uploader: {metadata.get('uploader', 'N/A')}")
            
            print('\n=== SUMMARY ===')
            print(f"Total Items: {result['totalItems']}")
            print(f"Videos: {result['videoCount']}")
            print(f"Audio: {result['audioCount']}")
            print(f"Images: {result['imageCount']}")
            
            print('\n=== ALL MEDIA ===')
            for index, item in enumerate(media, 1):
                print(f"\n[{index}] {item['type'].upper()}")
                print(f"  URL: {item['url'][:60]}...")
                print(f"  Quality: {item['quality']}")
                print(f"  Format: {item['format']}")
                if item.get('filesize'):
                    print(f"  Size: {item['filesize'] / 1024 / 1024:.2f} MB")
                if item.get('isBest'):
                    print("  ⭐ BEST QUALITY")
            
            return result
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

def download_by_type(url: str, media_type: str) -> dict:
    """
    Download media filtered by type
    
    Args:
        url: Any URL supported by yt-dlp
        media_type: Media type - 'video', 'audio', or 'image'
        
    Returns:
        dict: Filtered media data or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/alldl/type', params={'url': url, 'type': media_type})
        data = response.json()
        
        if data.get('success'):
            result = data['data']
            metadata = result['metadata']
            media = result['media']
            
            print('=== METADATA ===')
            print(f"Title: {metadata['title']}")
            print(f"Platform: {metadata['platform']}")
            
            print(f"\n=== {media_type.upper()} ONLY ({result['totalItems']} items) ===")
            for index, item in enumerate(media, 1):
                print(f"\n[{index}] {item['type'].upper()}")
                print(f"  URL: {item['url'][:60]}...")
                print(f"  Quality: {item['quality']}")
                print(f"  Format: {item['format']}")
                
                if media_type == 'video' and item.get('resolution'):
                    print(f"  Resolution: {item['resolution']}")
                    print(f"  FPS: {item.get('fps')}")
                    print(f"  Codec: {item.get('vcodec')}")
                
                if media_type == 'audio':
                    print(f"  Codec: {item.get('acodec')}")
                    print(f"  Bitrate: {item.get('abr')} kbps")
                    print(f"  Sample Rate: {item.get('asr')} Hz")
                
                if media_type == 'image' and item.get('width'):
                    print(f"  Resolution: {item['width']}x{item['height']}")
                
                if item.get('filesize'):
                    print(f"  Size: {item['filesize'] / 1024 / 1024:.2f} MB")
            
            return result
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

def get_best_quality(url: str) -> dict:
    """
    Get only the best quality video
    
    Args:
        url: Any URL supported by yt-dlp
        
    Returns:
        dict: Best quality media or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/alldl', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            media = data['data']['media']
            best = next((m for m in media if m.get('isBest')), None)
            
            if best:
                print('=== BEST QUALITY ===')
                print(f"Type: {best['type']}")
                print(f"Quality: {best['quality']}")
                print(f"Format: {best['format']}")
                print(f"Download URL: {best['url']}")
                
                return best
        return None
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

if __name__ == '__main__':
    # Example 1: Download all media
    print('=== EXAMPLE 1: ALL MEDIA ===\n')
    youtube_url = 'https://www.youtube.com/watch?v=dQw4w9WgXcQ'
    download_all_media(youtube_url)
    
    print('\n\n=== EXAMPLE 2: AUDIO ONLY ===\n')
    download_by_type(youtube_url, 'audio')
    
    print('\n\n=== EXAMPLE 3: IMAGES ONLY ===\n')
    download_by_type(youtube_url, 'image')
    
    print('\n\n=== EXAMPLE 4: BEST QUALITY ===\n')
    get_best_quality(youtube_url)
    
    print('\n\n=== EXAMPLE 5: OTHER PLATFORMS ===\n')
    
    # TikTok
    print('\n--- TikTok ---')
    download_all_media('https://www.tiktok.com/@user/video/123456789')
    
    # Vimeo
    print('\n--- Vimeo ---')
    download_all_media('https://vimeo.com/123456789')
    
    # SoundCloud
    print('\n--- SoundCloud ---')
    download_by_type('https://soundcloud.com/artist/track', 'audio')

File: examples/nodejs/alldl.js
const axios = require('axios');

const API_BASE = 'https://cog.api.br/api/v1';

/**
 * Download all available media from any URL
 * @param {string} url - Any URL supported by yt-dlp
 */
async function downloadAllMedia(url) {
  try {
    const response = await axios.get(`${API_BASE}/alldl`, {
      params: { url }
    });

    if (response.data.success) {
      const { metadata, media, totalItems, videoCount, audioCount, imageCount } = response.data.data;
      
      console.log('=== METADATA ===');
      console.log('Title:', metadata.title);
      console.log('Platform:', metadata.platform);
      console.log('Duration:', metadata.duration, 'seconds');
      console.log('Views:', metadata.views);
      console.log('Uploader:', metadata.uploader);
      
      console.log('\n=== SUMMARY ===');
      console.log('Total Items:', totalItems);
      console.log('Videos:', videoCount);
      console.log('Audio:', audioCount);
      console.log('Images:', imageCount);
      
      console.log('\n=== ALL MEDIA ===');
      media.forEach((item, index) => {
        console.log(`\n[${index + 1}] ${item.type.toUpperCase()}`);
        console.log('  URL:', item.url.substring(0, 60) + '...');
        console.log('  Quality:', item.quality);
        console.log('  Format:', item.format);
        if (item.filesize) {
          console.log('  Size:', (item.filesize / 1024 / 1024).toFixed(2), 'MB');
        }
        if (item.isBest) {
          console.log('  ⭐ BEST QUALITY');
        }
      });
      
      return response.data.data;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

/**
 * Download media filtered by type
 * @param {string} url - Any URL supported by yt-dlp
 * @param {string} type - Media type: 'video', 'audio', or 'image'
 */
async function downloadByType(url, type) {
  try {
    const response = await axios.get(`${API_BASE}/alldl/type`, {
      params: { url, type }
    });

    if (response.data.success) {
      const { metadata, media, totalItems } = response.data.data;
      
      console.log('=== METADATA ===');
      console.log('Title:', metadata.title);
      console.log('Platform:', metadata.platform);
      
      console.log(`\n=== ${type.toUpperCase()} ONLY (${totalItems} items) ===`);
      media.forEach((item, index) => {
        console.log(`\n[${index + 1}] ${item.type.toUpperCase()}`);
        console.log('  URL:', item.url.substring(0, 60) + '...');
        console.log('  Quality:', item.quality);
        console.log('  Format:', item.format);
        
        if (type === 'video' && item.resolution) {
          console.log('  Resolution:', item.resolution);
          console.log('  FPS:', item.fps);
          console.log('  Codec:', item.vcodec);
        }
        
        if (type === 'audio') {
          console.log('  Codec:', item.acodec);
          console.log('  Bitrate:', item.abr, 'kbps');
          console.log('  Sample Rate:', item.asr, 'Hz');
        }
        
        if (type === 'image' && item.width) {
          console.log('  Resolution:', `${item.width}x${item.height}`);
        }
        
        if (item.filesize) {
          console.log('  Size:', (item.filesize / 1024 / 1024).toFixed(2), 'MB');
        }
      });
      
      return response.data.data;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

/**
 * Get only the best quality video
 * @param {string} url - Any URL supported by yt-dlp
 */
async function getBestQuality(url) {
  try {
    const response = await axios.get(`${API_BASE}/alldl`, {
      params: { url }
    });

    if (response.data.success) {
      const { media } = response.data.data;
      const best = media.find(m => m.isBest);
      
      if (best) {
        console.log('=== BEST QUALITY ===');
        console.log('Type:', best.type);
        console.log('Quality:', best.quality);
        console.log('Format:', best.format);
        console.log('Download URL:', best.url);
        
        return best;
      }
    }
    return null;
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

// Example usage
(async () => {
  // Example 1: Download all media
  console.log('=== EXAMPLE 1: ALL MEDIA ===\n');
  const youtubeUrl = 'https://www.youtube.com/watch?v=dQw4w9WgXcQ';
  await downloadAllMedia(youtubeUrl);
  
  console.log('\n\n=== EXAMPLE 2: AUDIO ONLY ===\n');
  await downloadByType(youtubeUrl, 'audio');
  
  console.log('\n\n=== EXAMPLE 3: IMAGES ONLY ===\n');
  await downloadByType(youtubeUrl, 'image');
  
  console.log('\n\n=== EXAMPLE 4: BEST QUALITY ===\n');
  await getBestQuality(youtubeUrl);
  
  console.log('\n\n=== EXAMPLE 5: OTHER PLATFORMS ===\n');
  
  // TikTok
  console.log('\n--- TikTok ---');
  await downloadAllMedia('https://www.tiktok.com/@user/video/123456789');
  
  // Vimeo
  console.log('\n--- Vimeo ---');
  await downloadAllMedia('https://vimeo.com/123456789');
  
  // SoundCloud
  console.log('\n--- SoundCloud ---');
  await downloadByType('https://soundcloud.com/artist/track', 'audio');
})();

module.exports = {
  downloadAllMedia,
  downloadByType,
  getBestQuality
};
